movement/analysis/filtering.py

```diff
@@ -0,0 +1,141 @@
+from datetime import datetime
+from typing import Union
+
+import numpy as np
+import xarray as xr
+
+
+def interp_pose(
+    ds: xr.Dataset,
+    method: str = "linear",
+    limit: Union[int, None] = None,
+    max_gap: Union[int, None] = None,
+    inplace: bool = False,
+) -> Union[xr.Dataset, None]:
+    """
+    Fills in NaN values by interpolating over the time dimension.
+
+    Parameters
+    ----------
+    ds : xarray.Dataset
+        Dataset containing pose tracks, confidence scores, and metadata.
+    method : str
+        String indicating which method to use for interpolation.
+        Default is `linear`. See documentation for
+        `xarray.DataSet.interpolate_na` for complete list of options.
+    limit : int | None
+        Maximum number of consecutive NaNs to interpolate over.
+        `None` indicates no limit, and is the default value.
+    max_gap : TODO: Clarify the difference between `limit` & `max_gap`
+        The largest gap of consecutive NaNs that will be
+        interpolated over. The default value is `None`.
+    inplace: bool
+        If true, updates the provided DataSet in place and returns
+        `None`.
+
+    Returns
+    -------
+    ds_thresholded : xr.DataArray
+        The provided dataset (ds), where NaN values have been
+        interpolated over using the parameters provided.
+    """
+    # TODO: This method interpolates over confidence values as well.
+    #  -> Figure out whether this is the desired default behavior.
+    ds_interpolated = ds.interpolate_na(
+        dim="time", method=method, limit=limit, max_gap=max_gap
+    )
+
+    # Logging
+    if "log" not in ds_interpolated.attrs.keys():
+        ds_interpolated.attrs["log"] = []
+
+    log_entry = {
+        "operation": "interp_pose",
+        "method": method,
+        "limit": limit,
+        "max_gap": max_gap,
+        "inplace": inplace,
+        "datetime": str(datetime.now()),
+    }
+    ds_interpolated.attrs["log"].append(log_entry)
+
+    if inplace:
+        ds["pose_tracks"] = ds_interpolated["pose_tracks"]
+        ds["confidence"] = ds_interpolated["confidence"]
+        ds.attrs["log"] = ds_interpolated.attrs["log"]
+        return None
+    else:
+        return ds_interpolated
+
+
+def filter_confidence(
```

Comment:
Also, currently you provide users with the option to run interpolation after the filtering, which is very convenient.
However, this can only be done with the default interpolation parameters, so I'm ambivalnet towards it. Maybe it makes more sense to keep the two completely separate?

Reply:
I agree with this. I've now changed the function name and removed the `interpolate` parameter from this function altogether.